code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,961 @@
+                (self.samples.gy_right < 0)
+                | (self.samples.gy_right > self.screen_resolution[1])
+            ] = np.nan
+
+            fig, axs = plt.subplots(2, 1, figsize=(8, 6), sharex=True)
+
+            axs[0].plot(self.samples["gx_right"], label="gx_right")
+            axs[1].plot(self.samples["gy_right"], label="gy_right")
+            axs[1].set_xlabel("time")
+            axs[1].set_ylabel("x coordinate [pixels]")
+            axs[1].set_ylabel("y coordinate [pixels]")
+
+        elif eye == "left":
+            self.samples.gx_left[
+                (self.samples.gx_left < 0)
+                | (self.samples.gx_left > self.screen_resolution[0])
+            ] = np.nan
+            self.samples.gy_left[
+                (self.samples.gy_left < 0)
+                | (self.samples.gy_left > self.screen_resolution[1])
+            ] = np.nan
+
+            fig, axs = plt.subplots(2, 1, figsize=(8, 6), sharex=True)
+
+            axs[0].plot(self.samples["time"], self.samples["gx_left"], label="gx_left")
+            axs[0].set_ylabel("gx_left")
+            axs[1].plot(self.samples["time"], self.samples["gy_left"], label="gy_left")
+            axs[1].set_xlabel("timestamp")
+            axs[1].set_ylabel("pupil area [pixels]")
+            axs[0].set_xlim(self.samples["time"].iloc[0])
+
+        else:
+            print("Invalid eye argument")
+
+        if notebook:
+            plt.show()
+            return None
+
+        if save:
+            plt.savefig(os.path.join(path_save, filename))
+            plt.clf()
+            return os.path.join(path_save, filename)
+
+        return None
+
+    def plot_heatmap_coordinate_density(
+        self,
+        eye: str = "right",
+        notebook: bool = True,
+        save: bool = False,
+        path_save: str = ".",
+        filename: Optional[str] = None,
+    ) -> Optional[str]:
+        """
+        Plots a heatmap for eye tracking coordinates.
+
+        Parameters
+        ----------
+        eye : str, optional
+            Specifies whether to plot for the "right" or "left" eye.
+        notebook : bool, optional
+            If True, the plot is shown in the Jupyter notebook.
+        save : bool, optional
+            If True, the plot is saved to a file.
+        path_save : str, optional
+            Path to save the plot file.
+        filename : str, optional
+            Name of the saved plot file.
+
+        Returns
+        -------
+        Optional[str]
+            The path to the saved plot file.
+
+        Example
+        -------
+        DwiSession4.plot_heatmap_coordinate_density(eye="left", screen_resolution=(1024, 768))
+        """
+        
+        if filename is None:
+            filename = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_heatmap.pdf"
+
+        plt.rcParams["figure.figsize"] = [10, 6]
+        cmap = sns.color_palette("coolwarm", as_cmap=True)
+
+        if eye == "right":
```

Comment:
the left eye is not implemented, does it make sense to do this?

I would rather include an explicit check at the top of all methods:

``` Python
if eye != "right":
    raise NotImplementedError
```